src/test/datascience/notebook/helper.ts

```diff
@@ -253,119 +258,150 @@ export async function closeNotebooks(disposables: IDisposable[] = []) {
-    // KERNELPUSH: Re-enable
-    // const { vscodeNotebook, kernelProvider } = await getServices();
-    // // Wait for the active editor to come up
-    // await waitForCondition(async () => !!vscodeNotebook.activeNotebookEditor, 10_000, 'Active editor not a notebook');
-    // // Get the list of kernels possible
-    // const kernels = (await kernelProvider.provideKernels(
-    // vscodeNotebook.activeNotebookEditor!.document,
-    // new CancellationTokenSource().token
-    // )) as VSCodeNotebookKernelMetadata[];
-    // traceInfo(`Kernels found for wait search: ${kernels?.map((k) => `${k.label}:${k.id}`).join('\n')}`);
-    // // Find the kernel id that matches the name we want
-    // let id: string | undefined;
-    // if (criteria.labelOrId) {
-    // const labelOrId = criteria.labelOrId;
-    // id = kernels?.find((k) => (labelOrId && k.label === labelOrId) || (k.id && k.id == labelOrId))?.id;
-    // if (!id) {
-    // // Try includes instead
-    // id = kernels?.find((k) => (labelOrId && k.label.includes(labelOrId)) || (k.id && k.id == labelOrId))?.id;
-    // }
-    // }
-    // if (criteria.interpreterPath && !id) {
-    // id = kernels
-    // ?.filter((k) => k.selection.interpreter)
-    // .find((k) => k.selection.interpreter!.path.toLowerCase().includes(criteria.interpreterPath!.toLowerCase()))
-    // ?.id;
-    // }
-    // traceInfo(`Switching to kernel id ${id}`);
-    // // Send a select kernel on the active notebook editor
-    // void commands.executeCommand('notebook.selectKernel', { id, extension: JVSC_EXTENSION_ID });
-    // const isRightKernel = () => {
-    // if (!vscodeNotebook.activeNotebookEditor) {
-    // return false;
-    // }
-    // if (!vscodeNotebook.activeNotebookEditor.kernel) {
-    // return false;
-    // }
-    // if (vscodeNotebook.activeNotebookEditor.kernel.id === id) {
-    // traceInfo(
-    // `Found selected kernel id:label ${vscodeNotebook.activeNotebookEditor.kernel.id}:${vscodeNotebook.activeNotebookEditor.kernel.label}`
-    // );
-    // return true;
-    // }
-    // traceInfo(
-    // `Active kernel is id:label = ${vscodeNotebook.activeNotebookEditor.kernel.id}:${vscodeNotebook.activeNotebookEditor.kernel.label}`
-    // );
-    // return false;
-    // };
-    // await waitForCondition(
-    // async () => isRightKernel(),
-    // defaultTimeout,
-    // `Kernel with criteria ${JSON.stringify(criteria)} not selected`
-    // );
-    // // Make sure the kernel is actually in use before returning (switching is async)
-    // await sleep(500);
-}
-
-export async function waitForKernelToGetAutoSelected(_expectedLanguage?: string, _time = 100_000) {
-    // KERNELPUSH: Re-enable
-    // const { vscodeNotebook } = await getServices();
-    // // Wait for the active kernel to be a julia kernel.
+export async function waitForKernelToChange(criteria: { labelOrId?: string; interpreterPath?: string }) {
```

Comment:
These functions were commented out so I could get the fix out to customers. Uncommenting and replacing kernelProvider stuff with notebookControllerManager mostly.